pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -448,8 +400,19 @@ func (qt *QueueTracker) canRunApp(hierarchy []string, applicationID string, trac
 		} else if trackType == group {
 			config = m.getGroupWildCardLimitsConfig(qt.queuePath)
 		}
-		if config != nil && config.maxApplications != 0 && running > int(config.maxApplications) {
-			return false
+		if config != nil && config.maxApplications != 0 {
```

Comment:
See above comment we should not do this as part of the headroom check. We do it once as part of the queue tracker create and then depend on the config update to maintain it